### DISCUSSION TITLE
Serve without CLI

### DISCUSSION THREAD

**Discussion by oSethoum** [pocketbase/pocketbase#1684] [Q&A] [answered]
Am trying to use pocketbase as library within a golang app, am not sure how can I start the server without using the CLI, following the documentation I did, 
```go
package main

import (
	"github.com/pocketbase/pocketbase"
)

func main() {
	app := pocketbase.New()
	app.Start()
}

```
but this doesn't start the server, also how can I provide the config and migrate the schema within the go app without using the admin UI?

**Comment by ganigeorgiev** [comment, marked as answer]
If you want to start only the HTTP service, you can do something like:

```go
app := pocketbase.New()
app.Bootstrap()
serveCmd := cmd.NewServeCommand(app, false)
serveCmd.Execute()
```
(_more configuration options for the HTTP server will be available in the future_)

> how can I provide the config and migrate the schema within the go app without using the admin UI?

You can use migrations (there is also an option for automigrations when making changes from the UI).
More information how to setup the migrations you can find in https://pocketbase.io/docs/migrations/ (you don't need the migrate cmd if you don't want the cli interface; migrations will be applied automatically on serve).
More information on how to work with collections programatically you can find in https://pocketbase.io/docs/collection-methods/.
